src/oph/heratepalvelu/tep/jaksoHandler.clj

```diff
@@ -274,15 +269,14 @@
             (let [koulutustoimija (c/get-koulutustoimija-oid opiskeluoikeus)]
               (if (some? (tep-herate-checker herate))
                 (log/error {:herate herate :msg (tep-herate-checker herate)})
-                (when (and (c/check-opiskeluoikeus-tila
-                             opiskeluoikeus (:loppupvm herate))
-                           (check-not-fully-keskeytynyt herate)
-                           (c/check-opiskeluoikeus-suoritus-types?
-                             opiskeluoikeus)
-                           (not (c/feedback-collecting-prevented?
-                                  opiskeluoikeus
-                                  (:loppupvm herate)))
-                           (c/check-sisaltyy-opiskeluoikeuteen? opiskeluoikeus))
+                (when-not
+                 (or (c/terminaalitilassa? opiskeluoikeus (:loppupvm herate))
+                     (fully-keskeytynyt? herate)
+                     (not (c/has-one-or-more-ammatillinen-tutkinto?
```

Comment:
Mun mielestä yleensä sisäkkäiset not:t on pahasta ja siksi pitäisin tän mieluummin `when`- kuin `when-not`-rakenteena.  (Vaikka not:ien kokonaismäärä väheneekin.)